build/Common.prod.props

```diff
@@ -3,7 +3,12 @@
 
   <PropertyGroup>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
-    <DefaultTargetFrameworks>net6.0;netstandard2.0;net462</DefaultTargetFrameworks>
+    <TargetFrameworksForLibraries>net6.0;netstandard2.0;net462</TargetFrameworksForLibraries>
```

Comment:
since everything in prod.props is a Library anyway, we could use diff. name by avoiding "library"...
DefaultTargetFrameworks, DefaultTargetFrameworksExtended seems okay to me plus the
TargetFrameworksForAspNetCoreInstrumentation, etc..

Reply:
I feel the word "default" is generating confusion. The way I look at it is that:
1. There is no "default", all of them are explcit.
2. Having "TargetFrameworksForXyz" makes it more consistent.

Reply:
If you look at what I was doing in #4902, I consider this as "default" (the individual projects don't have to add TargetFrameworks at all since the props file covered it).

Reply:
I agree with @cijothomas I don't love "Libraries" in the name because all of these things are really libraries in some regard but I'm fine to leave as-is and we can continue to mess with it 😄